enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -0,0 +1,632 @@
+	// "PublicKey" indicates that the policy relies on a PGP publicKey and may optionally use a Rekor verification.
+	// "FulcioCAWithRekor" indicates that the policy is based on the Fulcio certification and incorporates a Rekor verification.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	PolicyType PolicyType `json:"policyType"`
+	// PublicKey defines the root of trust based on a PGP public key.
+	// +optional
+	PublicKey *PublicKey `json:"publicKey,omitempty"`
+	// FulcioCAWithRekor defines the root of trust based on the Fulcio certificate and the Rekor public key.
+  // For more information about Fulcio and Rekor, please refer to the document at:
+	// https://github.com/sigstore/fulcio and https://github.com/sigstore/rekor
+	// +optional
+	FulcioCAWithRekor *FulcioCAWithRekor `json:"fulcioCAWithRekor,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=PublicKey;FulcioCAWithRekor
+type PolicyType string
+
+const (
+	PublicKeyRootOfTrust         PolicyType = "PublicKey"
+	FulcioCAWithRekorRootOfTrust PolicyType = "FulcioCAWithRekor"
+)
+
+// PublicKey defines the root of trust based on a PGP public key.
+type PublicKey struct {
+	// keyData contains inline base64 encoded data of the public key.
+	// +kubebuilder:validation:Required
+	KeyData string `json:"keyData"`
+	// rekorKeyData contains inline base64 data of the Rekor public key.
+	// +optional
+	RekorKeyData string `json:"rekorKeyData,omitempty"`
+}
+
+// FulcioCAWithRekor defines the root of trust based on the Fulcio certificate and the Rekor public key.
+type FulcioCAWithRekor struct {
+	// fulcioCAData contains inline base64 data for the fulcio CA certificate.
+	// +kubebuilder:validation:Required
+	FulcioCAData string `json:"fulcioCAData"`
+	// rekorKeyData contains inline base64 data of the Rekor public key.
+	// +kubebuilder:validation:Required
+	RekorKeyData string `json:"rekorKeyData"`
+}
+
+// PolicyFulcioSubject defines the OIDC issuer and the email of the Fulcio authentication configuration.
+type PolicyFulcioSubject struct {
+	// oidcIssuer contains the expected OIDC issuer.
+	// Example: "https://expected.OIDC.issuer/"
+	// +kubebuilder:validation:Required
+	OIDCIssuer string `json:"oidcIssuer"`
+	// signedEmail holds the email address the the certificate is issued for.
+	// Example: "expected-signing-user@example.com"
+	// +kubebuilder:validation:Required
+	SignedEmail string `json:"signedEmail"`
+}
+
+// PolicyIdentity defines image identity the signature claims about the image. When omitted, the default matchPolicy is "MatchRepoDigestOrExact".
+// +kubebuilder:validation:XValidation:rule="(has(self.matchPolicy) && self.matchPolicy == 'ExactRepository') ? has(self.exactRepository) : true",message="must set exactRepository if matchPolicy is ExactRepository"
+// +kubebuilder:validation:XValidation:rule="(has(self.matchPolicy) && self.matchPolicy == 'RemapIdentity') ? has(self.remapIdentity) : true",message="must set remapIdentity if matchPolicy is RemapIdentity"
```

Comment:
The other unionmembers should be forbidden otherwise, see the examples in the API repo examples API group